crates/wasm/src/shared/conversion.rs

```diff
@@ -347,3 +348,120 @@ impl FromBindgen for wit::entity::AnimationActionStack {
         }
     }
 }
+
+impl FromBindgen for wit::types::Ulid {
+    type Item = Ulid;
+
+    fn from_bindgen(self) -> Self::Item {
+        Ulid::from(self)
+    }
+}
+
+impl IntoBindgen for Ulid {
+    type Item = wit::types::Ulid;
+
+    fn into_bindgen(self) -> Self::Item {
+        self.into()
+    }
+}
+
+macro_rules! procedural_storage_handle {
+    ($name:ident, $wit_name:ident) => {
+        impl FromBindgen for wit::$wit_name::Handle {
+            type Item = $name;
+
+            fn from_bindgen(self) -> Self::Item {
+                $name::from(self.ulid.from_bindgen())
+            }
+        }
+
+        impl IntoBindgen for $name {
+            type Item = wit::$wit_name::Handle;
+
+            fn into_bindgen(self) -> Self::Item {
+                wit::$wit_name::Handle {
+                    ulid: Ulid::from(self).into_bindgen(),
+                }
+            }
+        }
+    };
+}
+
+procedural_storage_handle!(ProceduralMeshHandle, client_mesh);
+procedural_storage_handle!(ProceduralTextureHandle, client_texture);
+procedural_storage_handle!(ProceduralSamplerHandle, client_sampler);
+procedural_storage_handle!(ProceduralMaterialHandle, client_material);
```

Comment:
Might be a good idea to use `paste!` so that you can define the valid handles in one place and share them across modules, a la `primitive_component_definitions`: https://github.com/AmbientRun/Ambient/blob/main/shared_crates/shared_types/src/lib.rs#L3

Not a blocker by any means though, but we may need to document how to add more procedural resources if it's not immediately obvious / there's no one source of truth

Reply:
TIL about `paste`. It did help to cleanup up these macros a lot, thanks!